airflow/api_connexion/endpoints/dataset_endpoint.py

```diff
@@ -120,3 +126,39 @@ def get_dataset_events(
     return dataset_event_collection_schema.dump(
         DatasetEventCollection(dataset_events=events, total_entries=total_entries)
     )
+
+
+@security.requires_access([(permissions.ACTION_CAN_CREATE, permissions.RESOURCE_DATASET)])
+@provide_session
+def post_dataset_event(session: Session = NEW_SESSION) -> APIResponse:
+    """Create an external dataset event. This endpoint is useful if you want to update a dataset and
+    trigger downstream DAG runs from external services.
+    """
+    try:
+        json_body = dataset_change_schema.load(get_json_request_dict())
+    except ValidationError as err:
+        raise BadRequest(detail=str(err))
+    uri = json_body["dataset_uri"]
+    external_source = json_body["external_source"]
```

Comment:
I don't think we should leave this open to the client, we should get this from the request, e.g. REMOTE_ADDR or registered api_client. Allowing this to be set by the clients puts us at a security risk as its integrity can't be verified anymore.